pkg/clusters/addons/kuma/addon.go

```diff
@@ -226,13 +226,23 @@ spec:
 // enableMTLS attempts to apply a Mesh resource with a basic retry mechanism to deal with delays in the Kuma webhook
 // startup
 func (a *Addon) enableMTLS(ctx context.Context, cluster clusters.Cluster) (err error) {
-	for i := 0; i < 12; i++ {
-		err = clusters.ApplyManifestByYAML(ctx, cluster, mtlsEnabledDefaultMesh)
-		if err != nil {
-			time.Sleep(time.Second * 5) //nolint:gomnd
-		} else {
-			break
+	timer := time.NewTicker(5 * time.Second) // nolint:gomnd
+	retryTimes := 0
+	maxRetries := 12
```

Comment:
What you could consider, refactoring this further, is to add a timer that's set to hit after a designated wait time instead relying on an implicit `maxRetries=12` and `tick` duration being 5 seconds.

This way you'll get more (arguably) idiomatic code with another select case below. 

WDYT?

Reply:
updated.